tiledb/sm/stats/stats.cc

```diff
@@ -244,17 +228,15 @@ void Stats::end_timer(const std::string& stat) {
 
 #else
 
-void Stats::add_counter(const std::string& stat, uint64_t count) {
-  (void)stat;
-  (void)count;
+void Stats::add_counter(const std::string&, uint64_t) {
 }
-ScopedExecutor Stats::start_timer(const std::string& stat) {
-  (void)stat;
-  return ScopedExecutor();
+
+DurationInstrument Stats::start_timer(const std::string&) {
```

Comment:
This is the alternate definition of `start_timer` for when `TILEDB_STATS` is false. As such it doesn't need to have the same return value as when stats is compiled it. The easiest thing to do is to have it return `int` here in that case. The benefit is to get closer to zero overhead. This will require putting an #ifdef in the header, in which case `start_timer` may be defined `inline` and `const` also and just return zero. It'll be optimized out in practice.

This might all be better done with policy templates, but this is a pretty simple case that doesn't benefit as much from that treatment.